inference/core/workflows/core_steps/models/foundation/google_vision_ocr/v1.py

```diff
@@ -0,0 +1,181 @@
+from typing import List, Literal, Optional, Type, Union
+from uuid import uuid4
+
+import numpy as np
+import requests
+import supervision as sv
+from pydantic import ConfigDict, Field
+from supervision.config import CLASS_NAME_DATA_FIELD
+
+from inference.core.workflows.execution_engine.constants import (
+    DETECTION_ID_KEY,
+    IMAGE_DIMENSIONS_KEY,
+    PREDICTION_TYPE_KEY,
+)
+from inference.core.workflows.execution_engine.entities.base import (
+    OutputDefinition,
+    WorkflowImageData,
+)
+from inference.core.workflows.execution_engine.entities.types import (
+    OBJECT_DETECTION_PREDICTION_KIND,
+    STRING_KIND,
+    StepOutputImageSelector,
+    WorkflowImageSelector,
+    WorkflowParameterSelector,
+)
+from inference.core.workflows.prototypes.block import (
+    BlockResult,
+    WorkflowBlock,
+    WorkflowBlockManifest,
+)
+
+LONG_DESCRIPTION = """
+Detect text in images using Google Vision OCR.
+
+Supported types of text detection:
+
+- `text_detection`: optimized for areas of text within a larger image.
+- `ocr_text_detection`: optimized for dense text documents.
+
+You need to provide your Google Vision API key to use this block.
+"""
+
+
+class BlockManifest(WorkflowBlockManifest):
+    model_config = ConfigDict(
+        json_schema_extra={
+            "name": "Google Vision OCR",
+            "version": "v1",
+            "short_description": "Detect text in images using Google Vision API",
+            "long_description": LONG_DESCRIPTION,
+            "license": "Apache-2.0",
+            "block_type": "model",
+        },
+        protected_namespaces=(),
+    )
+    type: Literal["roboflow_core/google_vision_ocr@v1"]
+    image: Union[WorkflowImageSelector, StepOutputImageSelector]
+    ocr_type: Literal["text_detection", "ocr_text_detection"]
```

Comment:
Apart from adding description here, you can provide:
```
Field(
        description="XXX",
        json_schema_extra={
            "values_metadata": {
                "text_detection": {"name": "Human Friendly Name", "description": "Short description"},
                "ocr_text_detection": {...},
             },
        },
    )
```
That would make this change in UI:
![image](https://github.com/user-attachments/assets/d3c290ee-fe7d-429b-813a-d98e326f76cb)
In descriptions I would explain the difference posted in google docs:
* `TEXT_DETECTION detects and extracts text from any image` - in a sense - "wild" scenes depicted in images
* `DOCUMENT_TEXT_DETECTION` is optimised for documents

Reply:
Done 👍